### ISSUE TITLE
Task list populated with list of facilities selected previously

### ISSUE THREAD

**Issue by suhafa** [bcgov/cas-reporting#265]
#### Description:

Now that we have a task list(#237), we need it to be populated with the list of facilities the user selects in 'Review facilities'
Note that for an SFO - there would only be 1. 

[Wireframe example](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=7826-16179&t=IlPsMexuLNx5JIwR-4)

Here is [what the screen will look like incase there wasn't any pre-selection of facilities](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=14765-5721&t=nGo5QWDD06hxJUss-1) earlier

#### Acceptance Criteria:

Given I am an Industry Reporter 
When I am filling out my operation's annual report
And I am on the Facilities Information step
Then I want to see the tasklist populated with the list of facilities I selected earlier so that I can report for each. 

Give I am a developer
When I am in Facilities Information 
Then I should be able, for each facility, to navigate the reporting steps and activities

#### Development Checklist:

- [ ] Update the task list component to include facility data.
- [ ] Ensure the task list updates dynamically based on user selections.
- [ ] Implement UI changes as per the provided wireframe.
- [ ]  Integrate facility data retrieval with the existing task list component.
- [ ] If no facility is selected then should show based on wireframe and save should be disabled.
- [ ] Ensure the task list supports single facility for SFO cases.
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Blocked By

#237 

#### Notes:

-

**Comment by patriciarussellCAS** [comment]
Hey Devs - In prep for next week's refinement, I'm wondering if someone would be willing to tackle the Dev Checklist for this ticket. @ayeshmcg @BallardRobinett or @joshgamache ?

**Comment by porton-devgroup** [comment]
Okay sure

**Comment by ayeshmcg** [comment]
Sure, can do that

**Comment by ayeshmcg** [comment]
@suhafa  @pbastia  I have added a Dev checklist for this. Could you please check to see if there is anything else that needs to be added?

**Comment by suhafa** [comment]
Looks good @ayeshmcg ! Here is [the wireframe for what it would look like if no facility was selected](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=14765-5721&t=nGo5QWDD06hxJUss-1) (this applies to LFOs, but this page and generic design can be used for other pages that depend on earlier pages to load the right content)

**Comment by patriciarussellCAS** [comment]
Is this blocked by #197 ?

**Comment by rdromey** [comment]
Design has changed, we're not putting facilities in the tasklist anymore.
